keras/callbacks.py

```diff
@@ -399,7 +399,7 @@ def on_epoch_end(self, epoch, logs=None):
             self.epochs_since_last_save = 0
             filepath = self.filepath.format(epoch=epoch, **logs)
             if self.save_best_only:
-                current = logs.get(self.monitor)
+                current = logs.get(self.monitor)[-1]
```

Comment:
I would disable it in these specific callbacks (throw a ValueError) because what the user wants to do is ambiguous. In that case it's better to let them subclass an existing callback.

And alternative would be to test for the type of `current` with `isinstance` and take the mean if it is not scalar. Unsure...